apps/voting/app/src/components/LocalIdentityBadge/LocalIdentityBadge.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Badge, IdentityBadge, font } from '@aragon/ui'
+import { IdentityContext } from '../IdentityManager/IdentityManager'
+
+function useIdentity(address) {
+  const [name, setName] = React.useState(null)
+  const { resolve, updates$, showLocalIdentityModal } = React.useContext(
+    IdentityContext
+  )
+
+  const handleNameChange = metadata => {
+    setName(metadata ? metadata.name : null)
+  }
+
+  const handleShowLocalIdentityModal = address => {
+    // Emit an event whenever the modal is closed (when the promise resolves)
+    return showLocalIdentityModal(address)
+      .then(() => updates$.next(address))
+      .catch(e => null)
+  }
+
+  React.useEffect(() => {
+    resolve(address).then(handleNameChange)
+
+    const subscription = updates$.subscribe(updatedAddress => {
+      if (updatedAddress.toLowerCase() === address.toLowerCase()) {
+        // Resolve and update state when the identity have been updated
+        resolve(address).then(handleNameChange)
+      }
+    })
+    return () => subscription.unsubscribe()
+  }, [address])
+
+  return [name, handleShowLocalIdentityModal]
+}
+
+const LocalIdentityBadge = ({ address, ...props }) => {
+  const [label, showLocalIdentityModal] = useIdentity(address)
+  const handleClick = () => showLocalIdentityModal(address)
+  return (
+    <IdentityBadge
+      {...props}
+      customLabel={label || ''}
+      entity={address}
+      popoverAction={{
+        label: `${label ? 'Edit' : 'Add'} custom label`,
+        onClick: handleClick,
+      }}
+      popoverTitle={
+        label ? (
+          <Wrap>
+            <Address>{label}</Address>
+            <StyledBadge>Custom label</StyledBadge>
+          </Wrap>
+        ) : (
+          'Address'
+        )
+      }
+    />
+  )
+}
+
+LocalIdentityBadge.propTypes = {
+  address: PropTypes.string.isRequired,
+}
+
+const Wrap = styled.div`
+  display: grid;
```

Comment:
Seems a _little_ bit unnecessary to use `grid` here 😄

Reply:
I don't see using `grid` as overkill over `flex` and I tend to use the first one before the latter by default. Let me know if you strongly think we'd need to use `flex` here.

Reply:
I'm not as familiar with the `grid` "language" as I am with the older layout ones, so that's partly why I'm not as comfortable using it everywhere.

But in general, my impression is that `grid` is so powerful you could probably build a spaceship with it, but it doesn't mean it's the _best_ or _easiest to maintain_ method of doing it :).

Reply:
Consider the verbosity levels here: if I were to use `flex`, I'd have to set the display on the container and then set `flex-grow` in one of the children with `grid` I get the same result with two rules on the container.